graf2d/gpadv7/inc/ROOT/RAttrAxis.hxx

```diff
@@ -19,153 +19,120 @@
-   bool IsLog2() const { return std::fabs(GetLog() - 2.) < 1e-6; }
-   bool IsLn() const { return std::fabs(GetLog() - 2.71828) < 0.1; }
-
-   RAttrAxis &SetSymlog(double const_value = 0.01)
-   {
-      if (const_value <= 0.)
-         fSymlog.Clear();
-      else
-         fSymlog = const_value;
-      return *this;
-   }
-   double GetSymlog() const { return fSymlog; }
-   bool HasSymlog() const { return fSymlog.Has(); }
-   void ClearSymlog() { fSymlog.Clear(); }
-
-   RAttrAxis &SetReverse(bool on = true) { fReverse = on; return *this; }
-   bool GetReverse() const { return fReverse; }
+class RAttrAxisLabels : public RAttrText {
+   R__ATTR_CLASS_DERIVED(RAttrAxisLabels, "labels", RAttrText)
 
-   RAttrAxis &SetTimeDisplay(const std::string &fmt = "", double offset = -1)
-   {
-      fTimeDisplay = true;
-      if (!fmt.empty()) fTimeFormat = fmt;
-      if (offset >= 0) fTimeOffset = offset;
-      return *this;
-   }
-   bool GetTimeDisplay() const { return fTimeDisplay; }
-   void ClearTimeDisplay()
-   {
-      fTimeDisplay.Clear();
-      fTimeOffset.Clear();
-      fTimeFormat.Clear();
-   }
+   RAttrValue<RPadLength> offset{this, "offset", {}};      ///<! labels offset - relative to "default" position
+   RAttrValue<bool> center{this, "center", false};         ///<! center labels
+   RAttrValue<bool> hide{this, "hide", false};             ///<! hide labels
+};
 
-   const RAttrLine &AttrLine() const { return fAttrLine; }
-   RAttrLine &AttrLine() { return fAttrLine; }
+/** \class RAttrAxisTitle
+\ingroup GpadROOT7
+\author Sergey Linev <s.linev@gsi.de>
+\date 2021-06-28
+\brief Axis title and its drawing attributes
+\warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback is welcome!
+*/
 
-   RAttrAxis &SetEndingSize(const RPadLength &sz) { fEndingSize = sz; return *this; }
-   RPadLength GetEndingSize() const { return fEndingSize; }
+class RAttrAxisTitle : public RAttrText {
+   R__ATTR_CLASS_DERIVED(RAttrAxisTitle, "title", RAttrText)
 
-   RAttrAxis &SetEndingStyle(const std::string &st) { fEndingStyle = st; return *this; }
-   RAttrAxis &SetEndingArrow() { return SetEndingStyle("arrow"); }
-   RAttrAxis &SetEndingCircle() { return SetEndingStyle("cicrle"); }
-   std::string GetEndingStyle() const { return fEndingStyle; }
+   RAttrValue<std::string> value{this, "value", ""};            ///<! axis title value
+   RAttrValue<std::string> position{this, "position", "right"}; ///<! axis title position - left, right, center
+   RAttrValue<RPadLength> offset{this, "offset", {}};           ///<! axis title offset - relative to "default" position
 
-   RAttrAxis &SetTicksSize(const RPadLength &sz) { fTicksSize = sz; return *this; }
-   RPadLength GetTicksSize() const { return fTicksSize; }
+   RAttrAxisTitle& operator=(const std::string &_title) { value = _title; return *this; }
```

Comment:
We discussed that for these special member names we'd spell usage as
```suggestion
   RAttrAxisTitle& operator=(const std::string &_title) { this->value = _title; return *this; }
```
to clarify that it's a member access, not access to a local variable. If you find this too much clutter for these simple cases then that's fine; I will call out cases below where this pattern is certainly more useful.